### ISSUE TITLE
Can't call `echoerr` from Lua

### ISSUE THREAD

**Issue by cbarrete** [neovim/neovim#13928]
- `nvim --version`: NVIM v0.5.0-dev+1086-g12f09cba3 (latest master)
- `vim -u DEFAULTS` (version: ) behaves differently? n/a
- Operating system/version: Debian
- Terminal name/version: Alacritty (irrelevant)
- `$TERM`: `alacritty` (irrelevant)

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC +"lua vim.cmd('echoe 3')"

```

### Actual behaviour

```
E5108: Error executing lua [string ":lua"]:1: Vim(echoerr):3
```

### Expected behaviour

Same as `:echoerr 3`

### Note

`echo` and `echom` work just fine.

**Comment by seandewar** [comment]
I believe `:echoerr` is being called, though the message is a tad verbose.

Errors thrown in certain contexts show slightly different messages.
For example, when inside VimL functions, you also get a special message along with the line number inside the function it originated from, e.g:
```
Error detected while processing function {function}:
line {line}:
{errmsg}
```

In Lua, errors are shown in a similiar format to what you've described, e.g:
```
E5108: Error executing lua {context}:{line}: {message}
```
You'll get a similiar format when calling Lua's `error()` directly.

With errors from VimL commands executed from Lua, the `{message}` bit is shown in a `Vim({command}):{errmsg}` format similiar to what's described in `:h :try` and `:h :catch`.
In other words, the bit you see after the last colon in the message is the error message from `:echoerr`.

**Comment by nyikos-zoltan** [comment]
I think this needs to be reopened, calling `echoerr` from lua (in any way) actually aborts lua execution. That is if you run this:
```lua
vim.cmd('echoerr "error"')
print('ok')
```
There is no "ok" message.
Looking into `:help :echoerr`, I think the problem is this:
> When used inside a try conditional, the message is raised as an error exception instead (see |try-echoerr|).

It seems like all lua code is running "inside a try conditional", causing this behavior.

**Comment by zeertzjq** [comment]
`vim.cmd()` is run "inside a try conditional".

**Comment by seandewar** [comment]
`vim.api.nvim_err_writeln()`

**Comment by nyikos-zoltan** [comment]
Oh, I didn't even think about looking for a `vim.api` method, thanks!
